src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -29653,6 +29653,158 @@ .maxstack  5
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("C.Main", """
+                {
+                  // Code size      145 (0x91)
+                  .maxstack  4
+                  .locals init (Base[] V_0,
+                                Base[] V_1,
+                                int V_2,
+                                Base[] V_3,
+                                System.ReadOnlySpan<Base> V_4,
+                                System.ReadOnlySpan<Base> V_5,
+                                System.Span<Base> V_6)
+                  IL_0000:  ldc.i4.1
+                  IL_0001:  newarr     "Derived"
+                  IL_0006:  dup
+                  IL_0007:  ldc.i4.0
+                  IL_0008:  newobj     "Derived..ctor()"
+                  IL_000d:  stelem.ref
+                  IL_000e:  stloc.0
+                  IL_000f:  ldloc.0
+                  IL_0010:  dup
+                  IL_0011:  ldc.i4.0
+                  IL_0012:  call       "void CollectionExtensions.Report(object, bool)"
+                  IL_0017:  dup
+                  IL_0018:  stloc.0
+                  IL_0019:  stloc.1
+                  IL_001a:  ldc.i4.0
+                  IL_001b:  stloc.2
+                  IL_001c:  ldloc.0
+                  IL_001d:  ldlen
+                  IL_001e:  conv.i4
+                  IL_001f:  ldloc.1
+                  IL_0020:  ldlen
+                  IL_0021:  conv.i4
+                  IL_0022:  add
+                  IL_0023:  newarr     "Base"
+                  IL_0028:  stloc.3
+                  IL_0029:  ldloca.s   V_4
+                  IL_002b:  ldloc.0
+                  IL_002c:  call       "System.ReadOnlySpan<Base>..ctor(Base[])"
+                  IL_0031:  ldloca.s   V_4
+                  IL_0033:  ldloc.3
+                  IL_0034:  newobj     "System.Span<Base>..ctor(Base[])"
+                  IL_0039:  stloc.s    V_6
+                  IL_003b:  ldloca.s   V_6
+                  IL_003d:  ldloc.2
+                  IL_003e:  ldloca.s   V_4
+                  IL_0040:  call       "int System.ReadOnlySpan<Base>.Length.get"
+                  IL_0045:  call       "System.Span<Base> System.Span<Base>.Slice(int, int)"
+                  IL_004a:  call       "void System.ReadOnlySpan<Base>.CopyTo(System.Span<Base>)"
+                  IL_004f:  ldloc.2
+                  IL_0050:  ldloca.s   V_4
+                  IL_0052:  call       "int System.ReadOnlySpan<Base>.Length.get"
+                  IL_0057:  add
+                  IL_0058:  stloc.2
+                  IL_0059:  ldloca.s   V_5
+                  IL_005b:  ldloc.1
+                  IL_005c:  call       "System.ReadOnlySpan<Base>..ctor(Base[])"
+                  IL_0061:  ldloca.s   V_5
+                  IL_0063:  ldloc.3
+                  IL_0064:  newobj     "System.Span<Base>..ctor(Base[])"
```

Comment:
>IL_0064:  newobj     "System.Span<Base>..ctor(Base[])" [](http://example.com/codeflow?start=18&length=54)

Here as well?

Reply:
We are creating a new "destination" span each time we copy a spread element to the resulting collection. Issue #71270 tracks improving this.